packages/shared/src/api/throttledClient.ts

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2018 - 2024 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { ExplorerProvider, NodeProvider } from '@alephium/web3'
+import pThrottle from 'p-throttle'
+
+import { defaultNetworkSettings } from '@/network'
+import { NetworkSettings } from '@/types/network'
+
+const throttle = pThrottle({
+  limit: 10,
+  interval: 1000
+})
+
+const throttledFetch = throttle((url, options = {}) => fetch(url, options))
```

Comment:
You would think that 

```ts
const throttledFetch = throttle(fetch)
```

is enough, but you'd be surprised to know how weird behavior this gives. I still don't fully understand why, but this works so there you go

Reply:
I had the exact same issue in the past! Same conclusion as yours :P